### ISSUE TITLE
New turf union returns a strange Polygon.

### ISSUE THREAD

**Issue by ilijaz** [Turfjs/turf#1392]
I am trying to unite two polygons.
First https://gist.github.com/ilijaz/e4fc41abb6e12fe9f346434a06d948db
Second https://gist.github.com/ilijaz/c7fb49ad134173ce6b502dbcc449ffa9
Also first polygon created after union of some as second polygons.
```
const union = require('@turf/union').default;
const turf = require('@turf/turf');

const first = {"..."};
const second = {"..."};
const un1 = union(first, second);
console.log(turf.area(un1), turf.area(first), turf.area(second))
```
I've had https://gist.github.com/ilijaz/51b426f9131ab29141bf2a6660e6ad77 as result. You can see it's wrong.
```
1334311.8920525517 20866.04727530979 268.7401291002204
```
Is it problem of tolerance? 
Is it normal that polygon has LinearRings that do not form a closed LlineString. They have only two coordinates.

Opposite of this case, union function from turf/turf works fine for me but it's about 2 times slowly. Here result on same input data https://gist.github.com/ilijaz/78997225ea130a0a1db1b40990f90237

**Comment by w8r** [comment]
your subject polygon is a multipolygon, instead of the polygon with a hole, so the union just consumes the second shape

**Comment by w8r** [comment]
https://codepen.io/w8r/pen/GdVbMz

**Comment by cglacet** [comment]
I can confirm that union is not working properly on version `6.0.2`. I'm too tired to look for the reason, but if you try with some examples on version `6.0.2` vs an older one, say `4.6` you'll see that they return pretty different polygons. If you want to try a quick thing, pick a random set of points, apply a Voronoi transform then pick a random subsets of the resulting polygons and compare different union versions.

Here is [one that fails on `6.0.2` (blue) vs. one that used to work on `4.6` (green)](http://geojson.io/#id=gist:cglacet/38020b3471610602a67e51ea067d8044&map=15/41.8933/12.4800) (All input feature polygons are in red-yellow).

I feel like it fails all the time, on this one it seems that he only takes into account 2 of the 5 input features.

**Comment by rowanwins** [comment]
Hi @cglacet 

In v6 onwards we're hoping to move to a new dependency for union operations (from jsts to martinez) as it's considerably smaller and faster.  Unfortunantly martinez doesn't work on feature collections, hence why it's only working on those two features and not all. 

Unfortunantly v6 got released a bit prematurely - I'm currently working on a v7 release where I hope to address to.

So for the timebeing stick with something in Turf v5 I think

**Comment by cglacet** [comment]
Ah ok.. maybe v5 should appear as the head version on npm then? Especially because the [live documentation](http://turfjs.org/docs#union) state that any number of polygons can be handled.

**Comment by rowanwins** [comment]
Yeah unfortunantly thing went a little haywire for a while - I'm doing my best to bring things back into order but for such a big repo it's taking a little while so patience is appreciated :)

**Comment by EvertEt** [comment]
I believe I have come accross the same behaviour since the code below returns a MultiPolygon instead of a Polygon with a hole. 
![image](https://user-images.githubusercontent.com/9197891/87058487-6e0b9080-c208-11ea-92fc-96a95d2fdb08.png)

```
union({"type":"Feature","geometry":{"type":"Polygon","coordinates":[[[-6.755905151367184,26.84183942462205],[3.3542633056640683,19.99528898482248],[17.734680175781257,31.538163813258336],[3.1654357910156294,10.339238082719248],[-6.755905151367184,26.84183942462205]]]},"properties":{"$ol_id":"5123ae52-2365-4308-a133-8f5cd8360c7b"}}, {"type":"Feature","geometry":{"type":"Polygon","coordinates":[[[-11.436767578124998,27.827539156401116],[21.274337768554688,27.87671441202157],[-11.436767578124998,23.703637072212516],[-11.436767578124998,27.827539156401116]]]},"properties":{}})
```

Can we expect v7 soon or should we downgrade back to the latest v5?

**Comment by w8r** [comment]
The underlying algorithm lib does not show this problem:
https://codepen.io/w8r/pen/ZEQRWog

**Comment by EvertEt** [comment]
Indeed. The bug is that here the algorithm assumes multiple polygons are separate but in this case they should not be split up as the second one is a hole in the first one. https://github.com/Turfjs/turf/blob/a371807f78364820f15ab8f9ee3dc5aba51b7bac/packages/turf-union/index.ts#L47

**Comment by rowanwins** [comment]
Resolved since v6.2.0
